packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -771,9 +796,40 @@ class IOSDevice extends Device {
       }
     }
     maxWaitForCI?.cancel();
+    await errorListener?.cancel();
     return localUri;
   }
 
+  /// Listen to device logs for crash on iOS 18.4+ due to JIT restriction. If
+  /// found, give guided error and throw tool exit. Returns null and does not
+  /// listen if device is less than iOS 18.4.
+  Future<StreamSubscription<String>?> _interceptErrorsFromLogs(
+    IOSApp? package, {
+    required DebuggingOptions debuggingOptions,
+  }) async {
+    // Currently only checking for kJITCrashFailureMessage, which only should
+    // be checked on iOS 18.4+.
+    if (sdkVersion == null || sdkVersion! < Version(18, 4, null)) {
+      return null;
+    }
+    final DeviceLogReader deviceLogReader = getLogReader(
+      app: package,
+      usingCISystem: debuggingOptions.usingCISystem,
+    );
+
+    final Stream<String> logStream = deviceLogReader.logLines;
+
+    final String deviceSdkVersion = await sdkNameAndVersion;
+
+    final StreamSubscription<String> errorListener = logStream.listen((String line) {
+      if (line.contains(kJITCrashFailureMessage)) {
```

Comment:
is the failure message the only thing in the crash log? or is there a better identifier (e.g. error code)?